### ISSUE TITLE
System.NullReferenceException on application close

### ISSUE THREAD

**Issue by sawo1337** [mbdavid/LiteDB#706]
Hi,

I have tried this with two separate machines, one with VS2013, another with 2015:

To reproduce this, I have created empty windows forms project with a single button in it. The button has the following code:

            using (var db = new LiteDatabase("MyData.db"))
            {
                // Upload a file from file system
                db.FileStorage.Upload("fileID", @"D:\path\system_logins.png");
            }
        }

The above executes fine, however, when closing the application, I get System.NullReferenceException from LiteDB.dll.
I have other weird nullreference exceptions which are only apparent with versions above 3.1.1, but this is the easiest to reproduce. The code I posted above is working just fine with version 3.1.1, but not newer versions.
I have also tried setting different .NET versions for my project, but none seems to work apart from going back to 3.1.1

**Comment by jmarcorb** [comment]
Weirdly enough, I just spent three hours with a NullReferenceException on similar terms. I have a project using an older LiteDB version(I do not remember, but it was the latest in May 2017), had to abandon it for a few months and now I just came back to it and updated the nuget packages.
The following code was returning a NullReferenceException from LiteDB:
`using (var db = new LiteDatabase("database.db")) { db.FileStorage.Upload(id, name, stream); }`
Rolling back to 3.1.1 as @sawo1337 did the trick for me.
Using VS2015.
Great job, by the way.

**Comment by MihaMarkic** [comment]
I have the same problem with 3.1.4. 
Here is the call stack
```
	LiteDB.dll!LiteDB.StreamExtensions.TryLock.AnonymousMethod__0()	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.FileHelper.TryExec(System.Action action, System.TimeSpan timeout)	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.StreamExtensions.TryLock(System.IO.FileStream stream, long position, long length, System.TimeSpan timeout)	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.FileDiskService.Lock(LiteDB.LockState state, System.TimeSpan timeout)	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.LockService.LockReserved()	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.LockService.Reserved()	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.LiteEngine.Transaction<int>(string collection, bool addIfNotExists, System.Func<LiteDB.CollectionPage, int> action)	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.LiteEngine.Update(string collection, System.Collections.Generic.IEnumerable<LiteDB.BsonDocument> docs)	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.LiteEngine.Update(string collection, LiteDB.BsonDocument doc)	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.LiteFileStream.Flush()	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.LiteFileStream.Dispose(bool disposing)	Unknown	No symbols loaded.
 	LiteDB.dll!LiteDB.LiteFileStream.~LiteFileStream()	Unknown	No symbols loaded.
```
Looks like this has been solved post 3.1.4.

**Comment by lbnascimento** [comment]
Hi! With the objective of organizing our issues, we are closing old unsolved issues. Please check the latest version of LiteDB and open a new issue if your problem/question/suggestion still applies. Thanks!
